### PR TITLE
Ensure that versions are always in sync

### PR DESCRIPTION
**Pull request by jdno**
When cutting a new release, three different versions need to be updated:

- The gemspec has the definitive version of the gem in it.
- The CHANGELOG contains the changes made in the different versions.
- The README file includes the current version in the installation
  instructions.

Not every developer is aware that these three places need to be changed,
and even then it is easy to miss one or two. To protect us against
inconsistencies, specs have been added that check that these three
versions exist, are in sync and meet certain requirements. The tests are
automatically run on CircleCI for each pull request to ensure that the
master branch is in a healthy state.

### CODE REVIEW

File: spec/versions/changelog_version_spec.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+RSpec.describe "Changelog version" do
+  def find_in_changelog(string)
+    File.open(changelog).each_line do |line|
+      return true if line.include? string
+    end
+
+    false
+  end
+
+  let(:changelog) { Pathname.new(__dir__).join("..", "..", "CHANGELOG.md") }
```
Comment (rjclaasen) [inline spec/versions/changelog_version_spec.rb]: What do you think about using `Rails.root` here? That way, this spec is independent of its own location in the file system.

File: spec/versions/readme_version_spec.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+RSpec.describe "README version" do
+  let(:readme) { Pathname.new(__dir__).join("..", "..", "README.md") }
```
Comment (rjclaasen) [inline spec/versions/readme_version_spec.rb]: & how about here?

File: spec/versions/changelog_version_spec.rb (hunk 1 above)
Comment (jdno) [reply spec/versions/changelog_version_spec.rb]: Since this is a gem and not a Rails application, there is no `Rails.root`. 😉

Comment (rjclaasen) [reply spec/versions/changelog_version_spec.rb]: Huh, looks like there is no way to get the root of an application in pure Ruby.

**Review by eboekhold — APPROVED** [review]
Nice stuff, looking good!

**Pull request merged**